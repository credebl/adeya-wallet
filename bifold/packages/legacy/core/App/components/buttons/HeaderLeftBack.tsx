import React from 'react'
import { StyleSheet, Text, TouchableOpacity, View } from 'react-native'
import Icon from 'react-native-vector-icons/MaterialCommunityIcons'

import { useTheme } from '../../contexts/theme'

const defaultIconSize = 38

interface HeaderLeftBackProps {
  title: string
  icon?: string
  testID?: string
  accessibilityLabel?: string
  size?: number
  onPress: () => void
  disabled?: boolean
}

const HeaderLeftBack: React.FC<HeaderLeftBackProps> = ({
  title,
  icon,
  size,
  testID,
  accessibilityLabel,
  onPress,
  disabled = false,
}) => {
  const accessible = accessibilityLabel && accessibilityLabel !== '' ? true : false
  const { ColorPallet, TextTheme } = useTheme()
  const style = StyleSheet.create({
    container: {
      flexDirection: 'row',
      alignItems: 'center',
    },
    touchableArea: {
      marginLeft: 10,
    },
    title: {
      ...TextTheme.normal,
      marginLeft: -4,
    },
  })

  return (
    <TouchableOpacity
      accessible={accessible}
      accessibilityLabel={accessibilityLabel}
      testID={testID}
      onPress={onPress}
      style={[style.touchableArea]}
      disabled={disabled}
    >
      <View style={style.container}>
        <Icon name={icon || 'chevron-left'} size={size || defaultIconSize} color={ColorPallet.brand.headerIcon} />
        <Text style={[style.title]}>{title}</Text>
      </View>
    </TouchableOpacity>
  )
}

export default HeaderLeftBack
