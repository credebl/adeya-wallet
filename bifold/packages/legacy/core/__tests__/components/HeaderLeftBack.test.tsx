import { render } from '@testing-library/react-native'
import React from 'react'

import HeaderLeftBack from '../../App/components/buttons/HeaderLeftBack'

describe('Header Left Button Component', () => {
  test('Renders correctly', () => {
    const tree = render(
      <HeaderLeftBack
        title={'Hello Header'}
        testID={'TestID.Button'}
        accessibilityLabel={'Hello'}
        onPress={() => {
          return
        }}
      />
    )

    expect(tree).toMatchSnapshot()
  })
})
