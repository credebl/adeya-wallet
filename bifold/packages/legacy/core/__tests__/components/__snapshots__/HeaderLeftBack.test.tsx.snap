// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Left Button Component Renders correctly 1`] = `
<View
  accessibilityLabel="Hello"
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "marginLeft": 10,
      "opacity": 1,
    }
  }
  testID="TestID.Button"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 38,
          },
          undefined,
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      Û∞ÖÅ
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 18,
            "fontWeight": "normal",
            "marginLeft": -4,
          },
        ]
      }
    >
      Hello Header
    </Text>
  </View>
</View>
`;
